### PR TITLE
Server certificate

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Added Support to Server Certificate for API 600, 800, 1000, and 1200.

### Issues Resolved
 N/A

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: libraries/resource_providers/api1200/synergy/1 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api1200/synergy/1]: please rename the filename.

File: libraries/resource_providers/api600/c7000/1 [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api600/c7000/1]: same as above

Comment (AsisBagga) [reply libraries/resource_providers/api600/c7000/1]: @soodpr  Corrected

File: libraries/resource_providers/api1200/synergy/1 (hunk 1 above)
Comment (AsisBagga) [reply libraries/resource_providers/api1200/synergy/1]: @soodpr Corrected

**Review by AnushaY1916 — APPROVED** [review]

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          @item.data['remoteIp'] = @item.data['name']
+          x = @item.get_certificate
+          @item.data.clear
+          @item.data['type'] = x['type']
+          @item.data['certificateDetails'] = []
+          @item.data['certificateDetails'][0] = {
+            'type' => x['certificateDetails'][0]['type'],
+            'base64Data' => x['certificateDetails'][0]['base64Data']
+          }
+          @item.data['name'] = @item.data['remoteIp']
```
Comment (soodpr) [inline libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: please check whats required fields are to pass in the import certificate request.

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          @item.data['aliasName'] = @item.data['remoteIp']
+          @item.data['uri'] = "/rest/certificates/servers/#{@item.data['aliasName']}"
+          @item.data.delete('remoteIp')
+          Chef::Log.info 'SSL_CERTIFCATE IMPORTED SUCCESSFULLY'
```
Comment (soodpr) [inline libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: this message should be printed after importing

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            @item.data.clear
+            @item.data['type'] = x['type']
+            @item.data['certificateDetails'] = []
+            @item.data['certificateDetails'][0] = {
+              'type' => x['certificateDetails'][0]['type'],
+              'aliasName' => alias_name,
+              'base64Data' => x['certificateDetails'][0]['base64Data']
+            }
+            @item.data['uri'] = "/rest/certificates/servers/#{alias_name}"
+            item.data.delete('remoteIp')
```
Comment (soodpr) [inline libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: we have already cleared the data..so no need for this line

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb (hunk 3 above)
Comment (AsisBagga) [reply libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: @soodpr Corrected

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb (hunk 4 above)
Comment (AsisBagga) [reply libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: @soodpr Corrected

File: libraries/resource_providers/api600/c7000/server_certificate_provider.rb (hunk 5 above)
Comment (AsisBagga) [reply libraries/resource_providers/api600/c7000/server_certificate_provider.rb]: @soodpr Corrected

**Review by soodpr — APPROVED** [review]

**Pull request merged**